packages/@webex/plugin-cc/src/services/task/index.ts

```diff
@@ -86,5 +86,92 @@ export default class Task implements ITask {
     }
   }
 
-  // TODO: Hold/resume, recording pause/resume, consult and transfer public methods to be implemented here
+  /**
+   * This is used to hold the task.
+   * @param taskId
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.hold(taskId).then(()=>{}).catch(()=>{})
+   * ```
+   * */
+  public async hold(taskId: string): Promise<TaskResponse> {
+    try {
+      return this.contact.hold({
+        interactionId: taskId,
+        data: {mediaResourceId: this.data.mediaResourceId},
+      });
+    } catch (error) {
+      const {error: detailedError} = getErrorDetails(error, 'hold', CC_FILE);
+      throw detailedError;
+    }
+  }
+
+  /**
+   * This is used to resume the task.
+   * @param taskId
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.resume(taskId).then(()=>{}).catch(()=>{})
+   * ```
+   */
+  public async resume(taskId: string): Promise<TaskResponse> {
+    try {
+      return this.contact.unHold({
+        interactionId: taskId,
+        data: {mediaResourceId: this.data.mediaResourceId},
+      });
+    } catch (error) {
+      const {error: detailedError} = getErrorDetails(error, 'resume', CC_FILE);
+      throw detailedError;
+    }
+  }
+
+  /**
+   * This is used to end the task.
+   * @param taskId - Unique Id to identify each task
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.end(taskId).then(()=>{}).catch(()=>{})
+   *  ```
+   */
+  public async end(taskId: string): Promise<TaskResponse> {
+    try {
+      return this.contact.end({interactionId: taskId});
+    } catch (error) {
+      const {error: detailedError} = getErrorDetails(error, 'end', CC_FILE);
+      throw detailedError;
+    }
+  }
+
+  /**
+   * This is used to wrap up the task.
+   * @param taskId - Unique Id to identify each task
+   * @param data - WrapupPayLoad
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.wrapup(taskId, data).then(()=>{}).catch(()=>{})
+   * ```
+   */
+  public async wrapup(taskId: string, wrapupPayload: WrapupPayLoad): Promise<TaskResponse> {
```

Comment:
I decided that we should accept the wrapup payload here - reason is it's very hard to get the agent config into this Task file, it is better if it is supplied by the external user. Further, the reason also needs to be supplied so I thought we can leave it open.
However, open to any other suggestions or changes.